shared/common-adapters/progress-bar.tsx

```diff
@@ -5,16 +5,27 @@ type Props = {
   ratio: number
   style?: Styles.StylesCrossPlatform
   fillStyle?: Styles.StylesCrossPlatform
+  flatRight?: boolean
+  flatLeft?: boolean
 }
 
-const ProgressBar = ({ratio, style, fillStyle}: Props) => {
+const ProgressBar = ({ratio, style, fillStyle, flatLeft, flatRight}: Props) => {
   const animatedStyles = {
     ...styles.inner,
     ...fillStyle,
+    ...(flatLeft ? styles.flatLeft : {}),
+    ...(flatRight ? styles.flatRight : {}),
     width: `${Math.max(0, Math.min(1, ratio)) * 100}%`,
   }
   return (
-    <Box style={Styles.collapseStyles([styles.outer, style])}>
+    <Box
+      style={Styles.collapseStyles([
+        styles.outer,
+        style,
+        flatLeft ? styles.flatLeft : {},
```

Comment:
`flatLeft && styles.flatLeft`. collapseStyles expects false and will maybe deopt you if you pass in new objects each time